src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -252,29 +233,37 @@ public void setMaximumPoolingConnections(int maximumPoolingConnections) {
     }
   }
 
-  /**
-   * (Re)initializes the HttpContext object. This removes any existing state (mainly cookies). Use an own
-   * cookie store, instead of the httpClient's global store, so cookies get reset on reinitialization
-   */
-  private void initializeHttpContext() {
-    CookieStore cookieStore = new BasicCookieStore();
-    httpContext = HttpClientContext.create();
+  protected HttpClientContext createHttpClientContext() {
+	HttpClientContext httpContext = HttpClientContext.create();
     httpContext.setCookieStore(cookieStore);
+    return httpContext;
   }
 
   @Override
   public void close() {
-    try {
-      httpClient.close();
-    } catch (IOException e) {
-      LOG.debug(e);
+    if(httpClient!=null) {
+	  synchronized(httpClientLock){
+	    if(httpClient!=null) {
```

Comment:
Maybe it will be better to move this code in something like IOUtils.closeQuietly to prevent copy-paste (and cleanup other places)?